awx/main/tests/functional/api/test_webhooks.py

```diff
@@ -202,3 +202,29 @@ def test_set_wrong_service_webhook_credential(organization_factory, job_template
     assert jt.webhook_key != ''
     assert jt.webhook_credential is None
     assert response.data == {'webhook_credential': ["Must match the selected webhook service."]}
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "service", [s for s, _ in WebhookTemplateMixin.SERVICES]
+)
+def test_unset_webhook_service_with_credential(organization_factory, job_template_factory, patch, service):
```

Comment:
I will try this out. The scenario I was trying to describe was an initial POST to create the job template with only a credential and no service, which worked

Reply:
so this is a different scenario (where the JT already existed with a service and a credential, but then the service is removed)

Reply:
Ok, I added another test a bit further above that captures this case.